<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav)"/>

  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb -ex run --args"/>

  <arg name="simulation" default="$(optenv SIMULATION false)" />
  <arg unless="$(arg simulation)" name="config_file" value="$(find uav_localize)/config/localization_params_F550.yaml"/>
  <arg     if="$(arg simulation)" name="config_file" value="$(find uav_localize)/config/sim_localization_params_F550.yaml"/>
  <arg unless="$(arg simulation)" name="cinfo_topic" value="/$(arg uav_name)/nerian_driver/depth_camera_info"/>
  <arg     if="$(arg simulation)" name="cinfo_topic" value="/$(arg uav_name)/realsense/camera/depth/camera_info/"/>

  <arg name="standalone" default="true" />
  <arg name="manager" default="localization_nodelet_manager" />
  <arg name="nodelet" value="uav_localize/LocalizeSingle" />
  <arg unless="$(arg standalone)" name="nodelet_load" value="load $(arg nodelet) $(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_load" value="standalone $(arg nodelet)"/>


  <group ns="$(arg uav_name)">
    
    <node pkg="nodelet" type="nodelet" name="uav_localization" args="$(arg nodelet_load)" output="screen" launch-prefix="$(arg launch_prefix)" >
      <param name="num_worker_threads" value="16" />
      
      <rosparam file="$(find uav_localize)/config/localization_params_common.yaml" />
      <rosparam file="$(arg config_file)" />
      <param name="uav_name" value="$(arg uav_name)" />
    
      <remap from="~detections" to="/$(arg uav_name)/uav_detection/detections" />
      <remap from="~trackings" to="/$(arg uav_name)/uav_tracking/trackings" />
      <remap from="~camera_info" to="$(arg cinfo_topic)" />

    </node>
  
  </group>

</launch>
